spec/furniture/marketplace/delivery_areas_controller_request_spec.rb

```diff
@@ -24,5 +32,6 @@
     end
 
     it { is_expected.to redirect_to(marketplace.location(child: :delivery_areas)) }
+    specify { expect { perform_request }.to change { marketplace.delivery_areas.reload.count }.by(1) }
```

Comment:
I was hoping to assert `label` and `price_cents` in the record. But 
`marketplace.delivery_areas.reload` returns an [ActiveRecord CollectionProxy](https://edgeapi.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html) which has no method that lets me dig into the record data in the collection.

So below attempts doesn't work and throw the expected `NoMethodError`.
```
specify { expect { perform_request }.to change { marketplace.delivery_areas.reload.label }.from(nil) }
```
When debugging w/ a `pry` breakpoint I noticed `marketplace.delivery_areas.reload.first.label` returns the data I want. But it doesn't work in the test code, throwing the same NoMethodError on the `ActiveRecord::Associations::CollectionProxy`

Reply:
You could use the [`exists?`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-exists-3F) predicate method with the `label` and `tax_rate` (i.e. `expect { perform_request }.to change { marketplace.delivery_areas.exists?(label: ..., tax_rate:...) }.from(false).to(true)` or something like that)

Reply:
If this is going to be a data comparison, I'd need to be able to pass the factory made data to the assertion. Maybe I'll  add `let`s for the `label` and `price_cents` in the `describe "#create"` block so I can use it in the `expect` block.

Reply:
Yea I tend to add a `let(:domain_model_attributes) { attributes_for(:domain_model) }` when I want to use the factory data elsewhere.